guides/learning_guide.rst

```diff
@@ -0,0 +1,1094 @@
+workload manager. Nearly all existing products were designed when the workflows
+were much simpler as shown in Figure 2a. Yet, these solutions have significant
+difficulties with emerging paradigms exemplified by Figure 2b, as well as with
+connecting and coordinating jobs. These problems have led users to develop their
+own ad hoc custom scheduling and resource management software or use tools that
+perform only workflow management or only scheduling. However, accomplishing 
+sufficient job coordination without first-class support from a workload manager
+like Flux has already proven to be difficult for either approach. Perhaps more
+importantly, developing and maintaining ad hoc management software—especially
+in this era of ever-evolving HPC hardware architectures—has quickly become
+prohibitively expensive for supercomputing application teams.
+
+.. figure:: images/figure-2.png
+   :alt: Different resource allocations between conventional and emerging scientific workflows
+   :align: center
+
+   Figure 2: An illustration of how the computing resources allocated to a job -- as granted by the HPC
+   workload manager -- differs between conventional and emerging scientific workflows. The
+   notional x-axes depict the compute node IDs allocated to the job and the y-axes depict the IDs of
+   computing resources in each of these nodes. (a) The conventional paradigm requires only a single
+   parallel simulation application to run. (b) The emerging paradigm often requires many different
+   types of tasks such as ensemble of molecular dynamics (MD) parallel simulation applications
+   and another ensemble of docking simulation applications along with in suti data analysis for the
+   MD ensemble while these tasks are driven by an AI.
+   
+
+Illustrative Batch Script Examples
+==================================
+
+In a job script used by a simpler workflow, the user enters line 1 in Figure 3a to
+submit a job script (line 2) to the system-level workload manager, requesting 256
+compute nodes with 42 CPU cores each. This job script uses the allocated resources
+under the conventional workflow paradigm (see Figure 2a), where the application
+(i.e., ``sim.app``) is executed across 256 compute nodes simultaneously, which then
+work together to perform a computationally challenging problem such as a physics
+simulation. More specifically, 42 copies of the ``sim.app`` software will be launcher and
+run on each of the 256 compute nodes, with one copy per CPU core.
+
+Figure 3b shows a job script with multiple tasks submitted on a heterogeneous HPC
+system. Notice this script still remains simple with Flux, requiring only a few more
+lines within the script. Similar to the previous example, at line 1, the user submits
+a job script (through lines 2–5) with a request of 256 compute nodes each with 42
+CPU cores—now also with 6 GPU resources (-g6). When this request is granted,
```

Comment:
inline literal `-g6`

Reply:
When I was writing this up all I could think of with "g6" was https://youtu.be/w4s6H4ku6ZY :laughing: